crates/editor/src/editor.rs

```diff
@@ -5323,7 +5327,7 @@ impl Editor {
         self.edit_prediction_settings =
             self.edit_prediction_settings_at_position(&buffer, cursor_buffer_position, cx);
 
-        if !self.edit_prediction_settings.is_enabled() {
+        if !self.edit_predictions_enabled() {
```

Comment:
Instead of incorporating a new field into this conditional, could we simply remove this early return? We already check this setting before *refreshing* the edit prediction (in `refresh_inline_completion`), so if `show_edit_predictions` is set to false, wouldn't `edits` already be  empty?

If not, I'd love to understand what I'm missing there.

Reply:
Yeah, you might be right. I went with the stateful approach because I thought it'd be good to keep the check so that predictions disappear if they are disabled for the current file from the menu. However, since yesterday, we discard active predictions as soon as the editor loses focus, so I guess it doesn't make a difference.